### PR TITLE
fix(avatar): remove box sizing for overlay text/icon

### PR DESCRIPTION
**Pull request by SheldonDialpad**
# PR Title

remove box sizing for overlay text/icon

## :hammer_and_wrench: Type Of Change

- [x] Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

In our application, there is `box-sizing` style in `commonextra.css` which impact all width calculation if there is border style. 
Screen Shot 2023-04-12 at 12 23 47 PM

This will cause our overlay text/icon fail to cover the avatar.
Screen Shot 2023-04-12 at 12 23 35 PM

To solve this issue, we need to force remove `box-sizing` style in overlay. So we add `d-box-unset` to overlay div.

## :pencil: Checklist

- [x] I have reviewed my changes
- [x] I have validated components with a screen reader
- [x] I have considered the performance impact of my change
- [x] I have checked that my change did not significantly increase bundle size

## :camera: Screenshots / GIFs

After remove `box-sizing` style in overlay div.
Screen Shot 2023-04-12 at 12 37 31 PM

## :link: Sources

This change will apply to https://dialpad.atlassian.net/browse/DP-66893

### CODE REVIEW

**Review by juliodialpad — CHANGES_REQUESTED** [review]
This seems pretty specific for your use-case, don't you think this could potentially break something while other users implement it?

Wouldn't it be better to add an `overlayClasses` prop so you can add custom classes to that overlay?

**Comment by juliodialpad** [comment]
Two more comments:

1. Make sure you're following the commit convention described here: https://github.com/dialpad/dialtone-vue/blob/staging/.github/COMMIT_CONVENTION.md - PR title should be something like: `fix(avatar): remove box sizing for overlay text/icon

2. In this PR: https://github.com/dialpad/dialtone-vue/pull/881 we forgot to add `overlayIcon` and `overlayText` props to `avatar_default.story.vue` so that props currently cannot be tested on the avatar page: https://vue.dialpad.design/deploy-previews/pr-906/?path=/story/components-avatar--initials&args=overlayText:
  - Can you please add that props to the file so we can properly test the behavior there?
  - Also add the list of icons instead of raw text for `overlayIcon` please, you can see an example of that implementation here: https://github.com/dialpad/dialtone-vue/blob/staging/components/icon/icon.stories.js#L15-L24

**Comment by iropolo** [comment]
On code perspective LGTM

File: components/avatar/avatar.vue
Comment (juliodialpad) [inline components/avatar/avatar.vue]:
```suggestion
        'd-bgc-black-900 d-o70 d-ps-absolute d-w100p d-h100p d-d-flex d-ai-center d-zi-base1 d-bar-circle',
```

The z-index needs to be 'd-zi-base1' in order to be able to override the avatarImage in certain cases.
Also we need d-bar-circle here so the overlay is always in circle shape

Before
Screenshot 2023-04-13 at 12 24 05 p m

After
Screenshot 2023-04-13 at 12 24 27 p m

File: recipes/list_items/contact_info/contact_info.vue
Comment (juliodialpad) [inline recipes/list_items/contact_info/contact_info.vue]:
```suggestion
              overlay-class="d-mn4 d-ba d-baw4 d-bc-white d-box-unset"
```

File: recipes/list_items/contact_info/contact_info.vue
Comment (juliodialpad) [inline recipes/list_items/contact_info/contact_info.vue]:
```suggestion
              overlay-class="d-mn4 d-ba d-baw4 d-bc-white d-box-unset"
```

**Review by juliodialpad — CHANGES_REQUESTED** [review]
Looks better now, just a couple issues I found.

Please replace this lines: 
- https://github.com/dialpad/dialtone-vue/blob/DP-66880-fix-box-sizing-issue/recipes/list_items/contact_info/contact_info_variants.story.vue#L263-L264
- https://github.com/dialpad/dialtone-vue/blob/DP-66880-fix-box-sizing-issue/recipes/list_items/contact_info/contact_info_variants.story.vue#L269-L270

e.g
```
{ src: 'static/recipes/list_items/contact_info/avatar.png' },
```

with a require statement, and provide the `initials` prop too, so we can get rid of all the errors

```
{ src: require('./avatar.png'), initials: 'initials' }
```

Those routes work well on local but once deployed it throws errors.

Screenshot 2023-04-13 at 12 40 20 p m

File: components/avatar/avatar.vue
Comment (SheldonDialpad) [reply components/avatar/avatar.vue]: > The z-index needs to be `d-zi-base1` in order to be able to override the avatarImage in certain cases.

Yeah, I thought as you before. However, we may face to multiple admins listen-in case later. If we use `d-zi-base1`, the overlay icon will cover next avatar.
Screen Shot 2023-04-14 at 8 50 18 AM

File: components/avatar/avatar.vue
Comment (juliodialpad) [inline components/avatar/avatar.vue]:
```suggestion
        'd-bgc-black-900 d-o70 d-ps-absolute d-w100p d-h100p d-d-flex d-ai-center d-bar-circle d-zi-base',
        this.overlayClass,
```

Update this after I merge #908 into staging

File: components/avatar/avatar.vue
Comment (juliodialpad) [inline components/avatar/avatar.vue:319]: Nice clean up, thanks

File: components/avatar/avatar.vue
Comment (juliodialpad) [inline components/avatar/avatar.vue:363]: Nice clean up, thanks!

File: components/avatar/avatar.vue
Comment (juliodialpad) [inline components/avatar/avatar.vue:372]: Lol, wonder why this was like this and no one noticed it.

**Review by juliodialpad — APPROVED** [review]
Looking good! just wait for my PR to be merged into staging and update before merging this.

**Pull request merged**
